src/components/Form/TalentLayerIdForm.tsx

```diff
@@ -44,20 +46,17 @@ function TalentLayerIdForm() {
       }),
   });
 
+    
   const onSubmit = async (
     submittedValues: IFormValues,
     { setSubmitting }: { setSubmitting: (isSubmitting: boolean) => void },
   ) => {
     if (account && account.address && account.isConnected && publicClient && walletClient) {
       try {
-        let tx;
-        const handlePrice: any = await publicClient.readContract({
-          address: config.contracts.talentLayerId,
-          abi: TalentLayerID.abi,
-          functionName: 'getHandlePrice',
-          args: [submittedValues.handle],
-          account: address,
-        });
+        let tx; 
+        const length = submittedValues.handle.length;
+        const handlePrice = length > 4 ? mintFee : shortHandlesMaxPrice / Math.pow(2, length - 1);
```

Comment:
it's not good to duplicate the logical several time, better to have directly as a function in the hook